aiokafka/group_coordinator.py

```diff
@@ -0,0 +1,847 @@
+            subscription (SubscriptionState): instance of SubscriptionState
+                located in kafka.consumer.subscription_state
+            group_id (str): name of the consumer group to join for dynamic
+                partition assignment (if enabled), and to use for fetching and
+                committing offsets. Default: 'kafka-python-default-group'
+            enable_auto_commit (bool): If true the consumer's offset will be
+                periodically committed in the background. Default: True.
+            auto_commit_interval_ms (int): milliseconds between automatic
+                offset commits, if enable_auto_commit is True. Default: 5000.
+            assignors (list): List of objects to use to distribute partition
+                ownership amongst consumer instances when group management is
+                used. Default: [RoundRobinPartitionAssignor]
+            heartbeat_interval_ms (int): The expected time in milliseconds
+                between heartbeats to the consumer coordinator when using
+                Kafka's group management feature. Heartbeats are used to ensure
+                that the consumer's session stays active and to facilitate
+                rebalancing when new consumers join or leave the group. The
+                value must be set lower than session_timeout_ms, but typically
+                should be set no higher than 1/3 of that value. It can be
+                adjusted even lower to control the expected time for normal
+                rebalances. Default: 3000
+            session_timeout_ms (int): The timeout used to detect failures when
+                using Kafka's group managementment facilities. Default: 30000
+            retry_backoff_ms (int): Milliseconds to backoff when retrying on
+                errors. Default: 100.
+        """
+        self._client = client
+        self._session_timeout_ms = session_timeout_ms
+        self._heartbeat_interval_ms = heartbeat_interval_ms
+        self._retry_backoff_ms = retry_backoff_ms
+        self.generation = OffsetCommitRequest.DEFAULT_GENERATION_ID
+        self.member_id = JoinGroupRequest.UNKNOWN_MEMBER_ID
+        self.group_id = group_id
+        self.coordinator_id = None
+        self.rejoin_needed = True
+        self.needs_join_prepare = True
+        self.loop = loop
+        # rejoin group can be called in parallel
+        # (from consumer and from heartbeat task), so we need lock
+        self._rejoin_lock = asyncio.Lock(loop=loop)
+        self._enable_auto_commit = enable_auto_commit
+        self._auto_commit_interval_ms = auto_commit_interval_ms
+        self._assignors = assignors
+        self._subscription = subscription
+        self._partitions_per_topic = {}
+        self._cluster = client.cluster
+        self._cluster.request_update()
+        self._cluster.add_listener(self._handle_metadata_update)
+        self._auto_commit_task = None
+        self._closing = asyncio.Future(loop=loop)
```

Comment:
Add a comment to signal, that we use it for fast closure, as waiters will be released on it